services/mfa/mfa.go

```diff
@@ -93,47 +90,51 @@ func (m *Mfa) Activate(user *model.User, token string) *model.AppError {
 
 	ok, err := otpConfig.Authenticate(trimmedToken)
 	if err != nil {
-		return model.NewAppError("Activate", "mfa.activate.authenticate.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return errors.Wrap(err, "unable to parse the token")
 	}
 
 	if !ok {
-		return model.NewAppError("Activate", "mfa.activate.bad_token.app_error", nil, "", http.StatusUnauthorized)
+		return &InvalidToken{}
 	}
 
-	if appErr := m.Store.User().UpdateMfaActive(user.Id, true); appErr != nil {
-		return model.NewAppError("Activate", "mfa.activate.save_active.app_error", nil, appErr.Error(), http.StatusInternalServerError)
+	if err := storeFunc(userID, true); err != nil {
+		return errors.Wrap(err, "unable to store mfa active")
 	}
 
 	return nil
 }
 
-func (m *Mfa) Deactivate(userId string) *model.AppError {
-	if err := m.checkConfig(); err != nil {
-		return err
-	}
+// Deactivate set the mfa as deactive, remove the mfa secret, store it with the StoreActive and StoreSecret functions provided
+func Deactivate(storeSecretFunc StoreSecret, storeActiveFunc StoreActive, userId string) error {
+	var wg sync.WaitGroup
+	wg.Add(2)
 
-	schan := make(chan error, 1)
+	var secretErr error
+	var activeErr error
```

Comment:
This is not a hot path by any means. Let's just make these synchronous.

Reply:
makes sense, lets do it synchronous.

Reply:
done